src/cpp/fastdds/rpc/ReplierImpl.cpp

```diff
@@ -357,53 +351,29 @@ ReplierImpl::RequesterMatchStatus ReplierImpl::requester_match_status(
            RequesterMatchStatus::PARTIALLY_MATCHED;
 }
 
-bool ReplierImpl::is_fully_matched() const
-{
-    PublicationMatchedStatus pub_status;
-    SubscriptionMatchedStatus sub_status;
-
-    if ((RETCODE_OK == replier_reader_->get_subscription_matched_status(sub_status)) &&
-            (RETCODE_OK == replier_writer_->get_publication_matched_status(pub_status)))
-    {
-        return (pub_status.current_count > 0) && (pub_status.current_count == sub_status.current_count);
-    }
-
-    EPROSIMA_LOG_ERROR(REPLIER, "Error getting matched subscriptions or publications");
-    return false;
-}
-
 ReplierImpl::RequesterMatchStatus ReplierImpl::wait_for_matching(
         const fastdds::dds::Duration_t& timeout,
-        const RequestInfo& info) const
+        const RequestInfo& info)
 {
-    WaitSet waitset;
-
-    // Add condition for changes in the replier reader matched status
-    StatusCondition& reader_condition = replier_reader_->get_statuscondition();
-    // Do not overwrite the original mask, since it is used by the server
-    StatusMask new_mask_reader, original_mask_reader = new_mask_reader = reader_condition.get_enabled_statuses();
-    new_mask_reader |= StatusMask::subscription_matched();
-    reader_condition.set_enabled_statuses(new_mask_reader);
-    waitset.attach_condition(reader_condition);
-
-    // Add condition for changes in the replier writer matched status
-    StatusCondition& writer_condition = replier_writer_->get_statuscondition();
-    // Do not overwrite the original mask, since it is used by the server
-    StatusMask new_mask_writer, original_mask_writer = new_mask_writer = writer_condition.get_enabled_statuses();
-    new_mask_writer |= StatusMask::publication_matched();
-    writer_condition.set_enabled_statuses(new_mask_writer);
-    waitset.attach_condition(writer_condition);
-
     Time_t current_time;
     Time_t::now(current_time);
     Time_t finish_time = current_time + timeout;
 
     RequesterMatchStatus requester_status = requester_match_status(info);
     while ((RequesterMatchStatus::MATCHED != requester_status) && current_time < finish_time)
     {
-        // Wait 10 ms for the conditions to be triggered
-        ConditionSeq active_conditions;
-        waitset.wait(active_conditions, {0, 10 * 1000 * 1000});
+        // Wait for the matched status to change
+        // Or every 100 milliseconds.
+        std::unique_lock<std::mutex> lock(mtx_);
+        cv_.wait_for(lock,
+                std::chrono::milliseconds(100),
+                [this]()
+                {
+                    return matched_status_changed_.load();
+                });
+
+        // Reset the matched status changed flag
+        matched_status_changed_.store(false);
 
         // Check if the replier is fully matched
         requester_status = requester_match_status(info);
```

Comment:
This should only be done if the wait call succeeds

Reply:
Well, that makes the `requester_match_status()` to be invoked more frequently after the first time that `matched_status_changed ` becomes `true` (maybe without any news in the status), but Im OK with it